[martilo] Spack exercise

lucia-agullo-marti opened the pull request:
# Summary
This pull request adds the `package.py` file for the Spack exercise project. It includes all necessary dependencies for versions 0.1.0, 0.2.0, and 0.3.0 of the software.

## Installation and usage
Install and use the packages, for example for version 0.1.0:

```bash
spack install spack-exercise@0.1.0
spack load spack-exercise@0.1.0
spackexample
```

Unload the package for example for version 0.1.0:

```bash
spack unload spack-exercise@0.1.0
```

ajaust reviewed (approved):
The Spack package and all package versions work as expected. :+1:

lucia-agullo-marti committed:
Add package.py file

This commit adds the package.py file for the Spack exercise, including the dependencies for all versions and also those for versions 0.2.0 and 0.3.0

IshaanDesai commented:
Ok :+1:
